File: api/src/models/imapemails.js
import Sequelize from 'sequelize';
import { sequelize } from '../database/database'
import User from './users';
import Email from './emails';


const ImapEmails =  sequelize.define('imapEmails', {
    id:{
        type:Sequelize.INTEGER,
        primaryKey:true,
        autoIncrement: true,
    },
    email_date:{
        type: Sequelize.STRING,
    },
    email_subject:{
        type: Sequelize.STRING
    },
    email_from:{
        type: Sequelize.STRING
    },
    email_to:{
        type: Sequelize.STRING
    },
    customer_id:{
        type: Sequelize.INTEGER
    },
    email_id:{
        type: Sequelize.INTEGER
    },
    email_uid:{
        type: Sequelize.INTEGER
    },
    email_date_o:{
        type: Sequelize.STRING
    },
    email_modseq:{
        type: Sequelize.STRING
    },
    x_gm_label:{
        type: Sequelize.STRING
    },
    x_gm_msgid:{
        type: Sequelize.STRING
    },
    x_gm_thrid:{
        type: Sequelize.STRING
    },
    email_seq:{
        type: Sequelize.STRING
    }
});

export default ImapEmails;
// User.belongsTo(Tasks, { foreignKey:'id', targetKey: 'customer_id'})
// Tasks.hasOne(User, { foreignKey:'id', targetKey: 'customer_id' }) 
//AccountGroup
User.hasMany(ImapEmails, { foreignKey: 'customer_id' });
ImapEmails.belongsTo(User, { foreignKey: 'customer_id', targetKey: 'id' });

File: ReadMail/ReadMail/Models/ImapEmails.cs
using System;
using System.Collections.Generic;

namespace ReadMail.Models
{
    public partial class ImapEmails
    {
        public int Id { get; set; }
        public DateTime? EmailDate { get; set; }
        public string EmailSubject { get; set; }
        public string EmailFrom { get; set; }
        public string EmailTo { get; set; }
        public int? EmailId { get; set; }
        public string EmailUid { get; set; }
        public string EmailDateO { get; set; }
        public string EmailModseq { get; set; }
        public string XGmLabel { get; set; }
        public string XGmMsgid { get; set; }
        public int? CustomerId { get; set; }
        public string EmailSeq { get; set; }
        public string XGmThrid { get; set; }
    }
}
